flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/debezium/DebeziumUtils.java

```diff
@@ -233,4 +235,79 @@ private static Map<String, String> querySystemVariables(
 
         return variables;
     }
+
+    public static BinlogOffset findBinlogOffset(long targetMs, MySqlConnection connection) {
+        MySqlConnection.MySqlConnectionConfiguration config = connection.connectionConfig();
+        BinaryLogClient client =
+                new BinaryLogClient(
+                        config.hostname(), config.port(), config.username(), config.password());
+
+        List<String> binlogFiles = new ArrayList<>();
+        JdbcConnection.ResultSetConsumer rsc =
+                rs -> {
+                    while (rs.next()) {
+                        String fileName = rs.getString(1);
+                        long fileSize = rs.getLong(2);
+                        if (fileSize > 0) {
+                            binlogFiles.add(fileName);
+                        }
+                    }
+                };
+
+        try {
+            connection.query("SHOW BINARY LOGS", rsc);
+            LOG.info("Total search binlog: {}", binlogFiles);
+
+            if (binlogFiles.isEmpty()) {
+                return BinlogOffset.ofBinlogFilePosition("", 0);
+            }
+
+            int index = searchBinlogOffset(client, targetMs, binlogFiles);
+            String binlogName = binlogFiles.get(index);
+            return BinlogOffset.ofBinlogFilePosition(binlogName, 0);
+        } catch (Exception e) {
+            throw new FlinkRuntimeException(e);
+        }
+    }
+
+    public static int searchBinlogOffset(
+            BinaryLogClient client, long targetMs, List<String> binlogFiles) {
+        int startIdx = 0;
+        int endIdx = binlogFiles.size() - 1;
+
+        while (startIdx <= endIdx) {
+            int mid = startIdx + (endIdx - startIdx) / 2;
+            FindOffsetListener midBinlog = getBinlogOffsetListener(client, binlogFiles.get(mid));
+            long midTs = midBinlog.getFirstEventTs();
+            if (midTs < targetMs) {
+                startIdx = mid + 1;
```

Comment:
I'm a little confused about which statement is correct? `startIdx=mid` or `startIdx = mix + 1`
Because the `getFirstEventTs` method would return a timestamp of the first event in the binglog file, Maybe the target timstamp is also in this binglog file?

Reply:
This case can be returned by endIdx, exp, startIdx=4, first timestamp=111, endIdx=5, first timestamp=222, our target timestamp=150, one the first loop, midTs= first timestamp of 4=111，startIdx=mid+1=5，we get second timestamp=222，and endIdx=mid-1=4，startIdx=5 > endIdx=4，end return endIdx=4

Reply:
This case can be returned by endIdx, exp, startIdx=4, first timestamp=111, endIdx=5, first timestamp=222, our target timestamp=150, one the first loop, midTs= first timestamp of 4=111，startIdx=mid+1=5，we get second timestamp=222，and endIdx=mid-1=4，startIdx=5 > endIdx=4，end return endIdx=4

Reply:
The index update logic looks good, and we should also deal with (print in logger and test in ut) the cases when the log corresponding to the timestamp was purged or has not yet been created.